lib/sqlalchemy/engine/_py_row.py

```diff
@@ -84,15 +64,13 @@ def __getstate__(self) -> Dict[str, Any]:
         return {
             "_parent": self._parent,
```

Comment:
**mike bayer** (zzzeek) wrote:

can confirm (visually) that pickle backwards compat is maintained here.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4575

Reply:
**Federico Caselli** (CaselIT) wrote:

so pickle compat is as follows:

older v2 pickle loads fine after this patch.
pickle saved in this patch does not load in older v2 code.

The issue is that we removed the key_style.

I think it's fine to keep the current logic (but maybe we mention it in the changelog)
Otherwise we could save a fixed number for the _key_style?

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4575